test/src/test_function.cpp

```diff
@@ -436,6 +436,19 @@ TEST(test_function, non_copyable_type)
     EXPECT_EQ(function.calls(), 2U);
 }
 
+// Test that non-copyable type refs work with stub
+TEST(test_function, non_copyable_type_ref)
+{
+    you_shall_not_copy yc;
+
+    stub::function<void(you_shall_not_copy&)> function;
+
+    function(std::move(yc));
```

Comment:
This code would not work, and also would not work with standard C++ functions:
http://cpp.sh/52gui

However, if the stub was: `stub::function<void(const you_shall_not_copy&)>` then that should work in a perfect world. But I think that also currently fails - is that something we should look into?